tests/ui/impl-trait/issue-72911.rs

```diff
@@ -15,7 +15,7 @@ fn gather_from_file(dir_entry: &foo::MissingItem) -> impl Iterator<Item = Lint>
 
 fn lint_files() -> impl Iterator<Item = foo::MissingItem> {
     //~^ ERROR: failed to resolve
-    unimplemented!()
+    std::iter::empty()
```

Comment:
why did you change this test?

Reply:
because it otherwise emits a "`(): Iterator` does not hold" error

Reply:
So this PR doesn't end up silencing errors, but also introducing new errors in some cases? 🤔 I think it may be important to note that.

Reply:
I reverted the change to show the diagnostic change